### PR TITLE
Quietly fall back to system ld for known cases where zld is not fully working

### PR DESCRIPTION
**Pull request by michaeleisel**
- arm64_32: Apple has not released the source code for this, and it's unclear if/when they will
- Catalyst: produces a spurious warning/error (depending if you have the recent tapi API PR applied)
- Xcode 12: uses tapi version 4, hopefully they'll release a version of tapi before Xcode 12 GM that solves this

The first one is pretty hard to fix without Apple deciding to open-source. The latter two I will figure out, if we have to, by just reverse-engineering the API.

One can force zld to be used through the new `-force_zld` flag however.

### CODE REVIEW

File: ld/src/ld/ld.cpp [hunk 1]
```diff
@@ -1336,9 +1342,39 @@ int main(int argc, const char* argv[])
...
+		for (int i = 0; i < argc; i++) {
+			if (strcmp(argv[i], "-force_zld") == 0) {
+				forceZld = true;
+			} else if (strcmp(argv[i], "-arch") == 0 && i < argc - 1 && strcmp(argv[i + 1], "arm64_32") == 0) {
+				isArm64_32 = true;
+			}
+		}
+
+		if (isArm64_32 && !forceZld) {
+            fallBackToSystemLd(argv, "WatchOS");
```
Comment (bjorn3) [inline ld/src/ld/ld.cpp:1357]:
```suggestion
                 fallBackToSystemLd(argv, "WatchOS");
```

**Comment by michaeleisel** [comment]
@rmaz @steipete @kastiglione @milend let me know if this will make life easier, and if there are any issues with the dependency on python and default of system ld

**Comment by kastiglione** [comment]
seems reasonable to me. For Xcode 12, it might be easier to configure Xcode to not use zld, instead of have zld recognize it's being called via Xcode 12.

**Comment by michaeleisel** [comment]
yeah, it's tricky. but, i think this system will have nice long-term benefits too, e.g. for xcode 13

**Comment by michaeleisel** [comment]
for consumers, note that zld_original_ld_path is also required, which if you're building with xcodebuild can just be `$(DT_TOOLCHAIN_DIR)/usr/bin/ld`. if you don't pass that flag, then behavior won't change from how it currently is

**Pull request merged**